CommonTools/PileupAlgos/plugins/PuppiPhoton.cc

```diff
@@ -123,6 +124,7 @@ void PuppiPhoton::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
         if(pupCol->refAt(iPF).key() != *itPho) continue;
         pWeight = weight_;
         if(!useValueMap_ && itPF->pt() != 0) pWeight = pWeight*(phoCands[iPho]->pt()/itPF->pt());
+	if(!useValueMap_ && itPF->pt() == 0) pVec.SetPxPyPzE(phoCands[iPho]->px()*pWeight,phoCands[iPho]->py()*pWeight,phoCands[iPho]->pz()*pWeight,phoCands[iPho]->energy()*pWeight);
```

Comment:
(workflow running AOD->miniAOD) looking event by event at saved MET vs sum(puppiMerged), it looks like this line is executed every event for every photon in the collection.
In the first 10 events I see roughly "-sum(puppiMerged.px) - sum(slimmedPhotons.px) ~ pfMetPuppi.px"

after removing this line, in most events "-sum(puppiMerged.px) = pfMetPuppi.px", except for a few (maybe one in 10) where there is difference.